packages/widgets-internal/farm/components/MerklNotice/index.tsx

```diff
@@ -12,15 +14,39 @@ type MerklNoticeContentProps = {
   linkColor?: string;
   merklLink: string;
   hasFarm?: boolean;
+  chainId?: number;
+  lpAddress?: Address;
 };
 
 export const MerklNoticeContent: React.FC<MerklNoticeContentProps> = ({
   linkColor = "primary",
   merklLink,
   hasFarm,
+  chainId,
+  lpAddress,
 }) => {
   const { t } = useTranslation();
 
+  const { data } = useQuery({
```

Comment:
For widgets we still want to keep it UI related only. Can we do the network IO in the web application and feed in the data to the component?

Reply:
sure, no problem

Reply:
Done